File: KoKoslan/src/java/eval/KoKoEvaluator.java
/**
 @author loriacarlos@gmail.com
 
*/
package kokoslan.ast;

public class KoKoEvaluator {

	private KoKoAst head;
	protected KoKoList args;

	public KoKoEvaluator(KoKoAst head, KoKoList args){
		this.head = head;
		this.args = args;
	}

	public KoKoValue evaluate(KoKoContext ctx){
		System.out.println("Vine a evaluar todo");
		String name = ((KoKoId)this.head).getValue();
        if(name.equals("print"))
            printArguments(ctx);
		return null;
	}

	public void printArguments(KoKoContext ctx){
        this.args
            .stream()
            .forEach( a -> System.out.println(a.eval(ctx)));
    }

	/*
		public KoKoValue evaluate(KoKoProgram tree, KoKoContext ctx){
			return null;
		}
	*/
	
}

File: KoKoslan/src/kotlin/eval/KoKoEvaluator.kt
/*
    Andrey Arguedas Espinoza
    Daniela Armas Sanchez
    Michael Chen Wang
    Kimberly Olivas Delgado
*/

package kokoslan.kt.eval

import kokoslan.kt.ast.*
import kokoslan.kt.exception.*

class KoKoEvaluator(private var head: KoKoAst, protected var args: KoKoList){

	fun evaluate(ctx: KoKoContext): KoKoValue?{
		val name: String = (this.head as KoKoId).getValue()
        if(name.equals("print"))
            printArguments(ctx)
		else{
            val kokoval = ctx.find (this.head as KoKoId)
            if(kokoval is KoKoLambdaValue) {
                var kl = kokoval as KoKoLambdaValue
                ctx.set(kl.pattern.toString(), ctx.getValue(args.get(0).toString()))
                return kl.expr.eval(ctx)
            }

        }
        return null
	}

	fun printArguments(ctx: KoKoContext){
        args.forEach{ print(it.eval(ctx)) }
    }

	/*
		fun evaluate(tree: KoKoProgram, ctx: KoKoContext): KoKoValue{
			return null;
		}
	*/
	
}